file/file_prefetch_buffer.cc

```diff
@@ -819,8 +819,20 @@ bool FilePrefetchBuffer::TryReadFromCacheUntracked(
       assert(reader != nullptr);
       assert(max_readahead_size_ >= readahead_size_);
 
+      // We disallow async IO for compaction reads since their
+      // latencies are not user-visible
```

Comment:
nit: ... are more tolerable ?? We have stats for user to know when compaction read is slow

Reply:
I can update the wording. The idea was that the compaction read was a background operation, in contrast to a latency-sensitive user-initiated scan operation